src/NuGet.Core/NuGet.Commands/PackagesLockFileBuilder.cs

```diff
@@ -46,7 +46,7 @@ public PackagesLockFile CreateNuGetLockFile(LockFile assetsFile)
 
                     var dependency = new LockFileDependency()
                     {
-                        Id = library.Name,
+                        Id = library.Name.ToLowerInvariant(),
                         ResolvedVersion = library.Version,
                         ContentHash = libraryLookup[identity].Sha512,
                         Dependencies = library.Dependencies
```

Comment:
Do dependencies should be lower-case?

From a nuget.exe restore in pacakges.config style I get:
```
      "microsoft.aspnet.webapi.client": {
        "type": "Direct",
        "requested": "[4.0.20710, )",
        "resolved": "4.0.20710",
        "contentHash": "BwFtW0cGr8tg02XalvF8MmznOpYrhquD2eXWI0oF3LLSqLMFoFkb8Pmke/7SnqkiNc+YfsaUqdXEKbjhv1Ctrw==",
        "dependencies": {
          "Microsoft.Net.Http": "[2.0.20710, 2.1.0)",
          "Newtonsoft.Json": "4.5.6"
        }
      },
```